src/ray/gcs/gcs_server/gcs_placement_group_scheduler.cc

```diff
@@ -242,33 +244,36 @@ void GcsPlacementGroupScheduler::ScheduleUnplacedBundles(
 
 void GcsPlacementGroupScheduler::DestroyPlacementGroupBundleResourcesIfExists(
     const PlacementGroupID &placement_group_id) {
-  bool is_committed = false;
-  bool is_prepared = false;
-  std::shared_ptr<BundleLocations> bundle_locations = std::make_shared<BundleLocations>();
+  std::shared_ptr<BundleLocations> committed_bundle_locations =
+      std::make_shared<BundleLocations>();
+  std::shared_ptr<BundleLocations> leasing_bundle_locations =
+      std::make_shared<BundleLocations>();
 
   // Check if we can find committed bundle locations.
   const auto &maybe_bundle_locations =
       committed_bundle_location_index_.GetBundleLocations(placement_group_id);
   // If bundle location has been already removed, it means bundles
   // are already destroyed. Do nothing.
   if (maybe_bundle_locations.has_value()) {
-    is_committed = true;
-    bundle_locations = maybe_bundle_locations.value();
+    committed_bundle_locations = maybe_bundle_locations.value();
   }
 
+  // Now let's see if there are leasing bundles. There could be leasing bundles and
+  // committed bundles at the same time if plaement groups are reshceduling.
   auto it = placement_group_leasing_in_progress_.find(placement_group_id);
   if (it != placement_group_leasing_in_progress_.end()) {
     const auto &leasing_context = it->second;
-    is_prepared = true;
-    bundle_locations = leasing_context->GetPreparedBundleLocations();
+    leasing_bundle_locations = leasing_context->GetPreparedBundleLocations();
   }
 
-  RAY_CHECK(!(is_committed && is_prepared))
```

Comment:
This is a problematic check. This can be true when some of already-committed bundles are rescheduled because nodes are dead.